From 980dec4a0f04a7133397ddd8bddb09270f4c5bfa Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 10 Apr 2017 17:26:48 +0800
Subject: [PATCH 1659/5242] MLK-14655-2 video: mxsfb: add default videomode
 for overlayfb

commit  305f59e4b28ce8e50ee86cbe0bdc838263ee7354 from
https://source.codeaurora.org/external/imx/linux-imx.git

Use the videomode of primary fb as the default videomode
for overlay fb during overlay fb initialization process.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit c3f12ccefc2025fbe4b9280e5715767f6dc72442)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 3c9d383..eae5264 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -2001,6 +2001,7 @@ static void init_mxsfb_overlay(struct mxsfb_info *fbi,
 	ofb->ol_fb->fbops = &overlay_fb_ops;
 	ofb->ol_fb->node  = -1;
 	ofb->ol_fb->par	  = ofb;
+	INIT_LIST_HEAD(&ofb->ol_fb->modelist);
 
 	ofb->id = 0;
 	ofb->ops = &ofb_ops;
@@ -2043,6 +2044,7 @@ static void mxsfb_overlay_init(struct mxsfb_info *fbi)
 {
 	int ret;
 	struct mxsfb_layer *ofb = &fbi->overlay;
+	struct fb_videomode ofb_vm;
 
 	ofb->dev = &fbi->pdev->dev;
 	ofb->ol_fb = framebuffer_alloc(0, ofb->dev);
@@ -2053,6 +2055,14 @@ static void mxsfb_overlay_init(struct mxsfb_info *fbi)
 
 	init_mxsfb_overlay(fbi, ofb);
 
+	/* add videomode to overlay fb */
+	fb_var_to_videomode(&ofb_vm, &fbi->fb_info->var);
+	ret = fb_add_videomode(&ofb_vm, &ofb->ol_fb->modelist);
+	if (ret) {
+		dev_err(ofb->dev, "add vm to ofb failed\n");
+		goto fb_release;
+	}
+
 	ret = register_framebuffer(ofb->ol_fb);
 	if (ret) {
 		dev_err(ofb->dev, "failed to register overlay\n");
-- 
1.7.9.5

